src/Data/Bifunctor/Linear.hs

```diff
@@ -23,14 +23,17 @@ import Data.Void
 -- * @'bimap' f g = 'first' f '.' 'second' g
 class Bifunctor p where
   {-# MINIMAL bimap | (first , second) #-}
-  bimap :: (a ->. b) -> (c ->. d) -> a `p` c ->. b `p` d
-  bimap f g = first f . second g
+  bimap :: (a ->. b) -> (c ->. d) -> a `p` c -> b `p` d
```

Comment:
Mmm wouldn't these be

```haskell
   bimap :: (a ->. b) -> (c ->. d) -> a `p` c -> b `p` d
```

I'd even expect `forall a. (Functor p a)` to be a super-constraint. But this last point can wait until another PR.

Reply:
I agree completely - this is what started the discussion above - I changed it back after your comment: 
> Looking concretely, we are considering `(,)` and `Either`, `Either` is not a Control bifunctor. So I'd say we can restrict our attention to Data Bifunctors for the time being. Hence 
```haskell
bimap :: (a ->. b) -> (c ->. d) -> a `p` c -> b`p` d
```

Reply:
I think that is was a typo on my part :slightly_frowning_face: . Quite unfortunate.